### PR TITLE
fix: work status sync when work dispatching is suspended

### LINKED ISSUE #5397: suspension dispatching vs scale subresource
**Please provide an in-depth description of the question you have**:
Scaling a Deployment in member-cluster for a Propagation Policy that has suspended dispatching still scales back to what the ResourceTemplate is. 

I need guidance on whether this is expected behavior or whether it needs to be fixed (and where). Providing steps below:

1. Quick start / Context
```
./hack/local-up-karmada.sh

export KUBECONFIG=/Users/amiralavi/.kube/karmada.config
kubectl config use-context karmada-apiserver
```

2. Create a PropagationPolicy

```
cat <<EOF | kubectl apply -f -
apiVersion: policy.karmada.io/v1alpha1
kind: PropagationPolicy
metadata:
  name: example-policy
spec:
  resourceSelectors:
    - apiVersion: apps/v1
      kind: Deployment
      name: nginx
  placement:
    clusterAffinity:
      clusterNames:
        - member1
EOF
````

3. Create a Deployment with 1 replica
```
kubectl create deployment nginx --image nginx
```

4. Observe that you have a Deployment with 1 replica
```
NAME    READY   UP-TO-DATE   AVAILABLE   AGE
nginx   1/1     1            1           13m
```

5. Suspend dispatching on PropagationPolicy
```
kubectl patch pp example-policy --type='json' -p='[{"op": "add", "path": "/spec/suspension", "value": {dispatching: true}}]'
```

6. Observe that ResourceBinding and Work are suspended
```
kubectl get rb nginx-deployment -ojsonpath='{.spec.suspension}'
{"dispatching":true}
```

```
kubectl get work -n karmada-es-member1  nginx-687f7fb96f -ojsonpath='{.spec.suspendDispatching}'
true
```

7. Scale deployment in member cluster
```
export KUBECONFIG=/Users/amiralavi/.kube/members.config
kubectl config use-context member1

kubectl scale deploy/nginx --replicas=2
```

8. Observe that it scales back down to 1
```
kubectl get deploy
NAME    READY   UP-TO-DATE   AVAILABLE   AGE
nginx   1/1     1            1           16m
```

```
kubectl get events

79s         Normal   ScalingReplicaSet         deployment/nginx              Scaled up replica set nginx-77b4fdf86c to 2 from 1
79s         Normal   ScalingReplicaSet         deployment/nginx              Scaled down replica set nginx-77b4fdf86c to 1 from 2
```

**What do you think about this question?**:

**Environment**:
- Karmada version: latest on main
- Kubernetes version: `1.27.3` -- what comes with quick-start
- Others:

### PR DESCRIPTION
**Pull request by a7i**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

As discussed in https://github.com/karmada-io/karmada/issues/5397 , we should not sync work status, because it attempts to update the resource in the member cluster

**Which issue(s) this PR fixes**:
Fixes #5397

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Thanks @a7i 
/kind feature
/retest

**Comment by a7i** [comment]
/retest

it failed on uploading logs 
Screenshot 2024-08-19 at 10 31 37 PM

File: pkg/controllers/status/work_status_controller.go [hunk 1]
```diff
@@ -222,6 +222,11 @@ func (c *WorkStatusController) syncWorkStatus(key util.QueueKey) error {
 		return nil
 	}
 
+	// Work in a suspend state means it never dispatches updates to the member cluster.
+	if helper.IsWorkSuspendDispatching(workObject) {
```
Comment (XiShanYongYe-Chang) [inline pkg/controllers/status/work_status_controller.go:226]: Hi @a7i If we add this judgment to this position, we also discard the state collection.

Comment (a7i) [reply pkg/controllers/status/work_status_controller.go:226]: good catch, updated.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks a lot~
/lgtm
/assign @RainbowMango

File: pkg/controllers/status/work_status_controller.go [hunk 2]
```diff
@@ -222,12 +222,25 @@ func (c *WorkStatusController) syncWorkStatus(key util.QueueKey) error {
...
 
+	if err := c.updateResource(ctx, observedObj, workObject, fedKey); err != nil {
+		return err
+	}
+
+	klog.Infof("Reflecting the resource(kind=%s, %s/%s) status to the Work(%s/%s).", observedObj.GetKind(), observedObj.GetNamespace(), observedObj.GetName(), workNamespace, workName)
+	return c.reflectStatus(ctx, workObject, observedObj)
+}
+
+func (c *WorkStatusController) updateResource(ctx context.Context, observedObj *unstructured.Unstructured, workObject *workv1alpha1.Work, fedKey keys.FederatedKey) error {
```
Comment (RainbowMango) [inline pkg/controllers/status/work_status_controller.go:233]: It looks great to have this method dedicated to updating. Thanks.

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**